### ISSUE TITLE
Handle connected sockets across horizontal scaling

### ISSUE THREAD

**Issue by mattkrick** [ParabolInc/parabol#3259]
Strategy: When an instance shuts down & when it starts up, all the websocket connections attached to that node should be wiped out from the cache since those connections are going to be closed. It has to happen on shutdown if we permanently shut down an instance (e.g. downscale). It has to happen on startup if an instance shuts down unexpectedly (e.g. power failure).

AC
- [ ] hook created that can be run on instance shutdown & instance startup that clears the `connectedSockets` array for each user
- [ ] get a guaranteed persisted ID from kubernetes that persists across restarts (e.g. instance N1, N2, etc)
- [ ] created `Connection` table in rethinkdb with `userId, nodeId, socketId`. index on userId, nodeId

 
Effort: 11 points

**Comment by mattkrick** [comment]
this needs to be a lot faster, ideally only in redis.

so,
- each team has a `team:${teamId}:online` set, if one isn't made, we create it
- when a user logins in, we check their tms and for each teamId, they get put in that set (userId::socketId)
- we grab all the members of that set and announce that the user has logged in
- if the user already exists in the set, we do nothing, but that should never happen since theyll have a unique socketId attached to the end of their userId
- when a user logs out, we look at their tms & grab the set for each
- we remove their userId::socketId & see if they have another one left
- if the userId is completely out of the set, then we announce to the rest that the userId is now offline
- if the set is empty, we delete it
- when a team is created, we create that set with the userId::socketId at the first entry
-  when a team is archived, we remove that set

**Comment by jordanh** [comment]
Scrubbed, let's keep it on the backlog-we'll want to get this done in the next 6 months or so

**Comment by jordanh** [comment]
Ah yes, this ol' chestnut. Good candidate for the next sprint. Going to prioritize

**Comment by mattkrick** [comment]
```ts
const redis = {
  'presence:user1': [{serverId: 'abc', socketId: 'foo3', lastSeenAtURL: '/meet/rew345'}, {serverId: 'def', socketId: 'foo8', lastSeenAtURL: '/me'}],
  'presence:user2': [{serverId: 'abc', socketId: 'foo4', lastSeenAtURL: '/meet/rew345'}],
  'onlineTeam::teamId123': ['user1', 'user2']
}
```

**Comment by mattkrick** [comment]
closed in favor of #4080
